src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -2583,18 +2583,30 @@ internal override NamedTypeSymbol TryLookupForwardedMetadataTypeWithCycleDetecti
                 {
                     var peModuleSymbol = (Metadata.PE.PEModuleSymbol)_modules[i];
 
-                    var forwardedToAssembly = peModuleSymbol.GetAssemblyForForwardedType(ref emittedName);
-                    if ((object)forwardedToAssembly != null)
+                    (AssemblySymbol firstSymbol, AssemblySymbol secondSymbol) = peModuleSymbol.GetAssembliesForForwardedType(ref emittedName);
+
+                    if ((object)firstSymbol != null)
                     {
+                        if ((object)secondSymbol != null)
+                        {
+                            var forwardingErrorInfo = new DiagnosticInfo(
+                                MessageProvider.Instance,
+                                (int)ErrorCode.ERR_TypeForwardedToMultipleAssemblies,
+                                emittedName.FullName,
+                                firstSymbol.Name,
+                                secondSymbol.Name);
```

Comment:
Consider passing the symbols as arguments rather than just the names, so the diagnostic contains fully-qualified names.